tpm2tools/signer.go

```diff
@@ -30,9 +30,13 @@ func (signer *tpmSigner) Public() crypto.PublicKey {
 // The opts hash function must also match the keys scheme.
 // Concurrent use of Sign is thread safe, but it is not safe to access the TPM
 // from other sources while Sign is executing.
+// For RSAPSS signatures, you cannot specify custom salt lengths. The salt
+// length will be (keyBits/8) - digestSize - 2, unless that is less than the
+// digestSize in which case, saltLen will be digestSize. The only normal case
+// where saltLen is not digestSize is when using 1024 keyBits with SHA512.
 func (signer *tpmSigner) Sign(_ io.Reader, digest []byte, opts crypto.SignerOpts) (signature []byte, err error) {
 	if _, ok := opts.(*rsa.PSSOptions); ok {
-		return nil, fmt.Errorf("signing with PSS not supported")
+		return nil, fmt.Errorf("signing with non-default PSS options not supported")
```

Comment:
I think we need more here. When we want an RSA signature, I think we still pass [PSSOptions](https://golang.org/pkg/crypto/rsa/#PSSOptions). What we really want to do is just check that:

 - The key gives us a PSS signature if we get a `PSSOptions`
 - `SaltLength` in the options matches what we expect based on the key.
   - I think that means checking that it's [`PSSSaltLengthAuto`](https://golang.org/pkg/crypto/rsa/#pkg-constants)